src/app/shared/services/authentication.service.ts

```diff
@@ -1,31 +1,57 @@
-import { Injectable } from '@angular/core';
+import { Injectable, OnDestroy } from '@angular/core';
 import { HttpClient } from '@angular/common/http';
 import { Router } from '@angular/router';
 
-import { BehaviorSubject, Observable, Subject } from 'rxjs';
+import { BehaviorSubject, Observable, Subject, concatMap, map } from 'rxjs';
 
 import { HttpUser } from '../interfaces/http-user.interface';
 
 @Injectable({
   providedIn: 'root'
 })
-export class AuthenticationService {
+export class AuthenticationService implements OnDestroy {
 
   loggedUser!: HttpUser | null;
   userSubject = new Subject<HttpUser>();
   isUserLogged = new Subject<boolean>();
   isAuthSubject = new BehaviorSubject<boolean>(false);
   isAuthenticated$: Observable<boolean> = this.isAuthSubject.asObservable();
   private apiUrl = 'http://localhost:3004';
+  private destroy$ = new Subject<void>();
 
   constructor(private http: HttpClient, private router: Router) {}
 
+  ngOnDestroy(): void {
+    this.destroy$.next();
+  }
```

Comment:
What is the purpose of this?

Reply:
The purpose is to unsubscribe in `onDestroy` hook to prevent memory leak. It unsubscribes when service will be closed (destroyed). In some sources  ngOnDestroy could be seen as `
ngOnDestroy() {
    this.destroyed$.next(true);
    this.destroyed$.complete();
  }`, but in other sources just without `this.destroyed$.complete()`